File: apps/codemap-search/experiments/event-navigation-poc/language-corpus/regressions/examples/python/controls/boundaries.py
class Router:
    def install(self, f, g):
        self.cb = f  # @S1
        self.other = g  # @S2
    def fire(self):
        fn = getattr(self, "cb")
        fn()  # @I1
    def custom(self):
        getattr = lambda obj, name: lambda: None
        fn = getattr(self, "cb")
        fn()  # @I2

File: apps/codemap-search/experiments/event-navigation-poc/language-corpus/regressions/examples/php/controls/boundaries.php
<?php
class Router {
    private $cb;
    private $other;
    private $name;
    function install($f, $g) {
        $this->cb = $f; // @S1
        $this->other = $g; // @S2
        $this->name = $f; // @S3
    }
    function fire() {
        $name = 'cb';
        ($this->$name)(); // @I1
    }
    function unresolved($name) {
        ($this->$name)(); // @I2
    }
}
